### ISSUE TITLE
Design/Wireframes needed in order to build UI

### ISSUE THREAD

**Issue by realslimshanky** [UdacityMobileWebScholarship/showcase-app#5]
We're in need of designs or wireframes in order to start the web app. Here are some suggestions using which a person can contribute.

- A landing page for the web app.
- A web-app submission page which takes the source code and other information about the project from the project.
- A showcase page which will include all the submissions.
- A dashboard for project submitter, project admin or with similar functionalities.

These are rough ideas and are open to discussions/suggestions.

**Comment by realslimshanky** [comment]
@divyamrastogi @PrabhanshuAttri @palnabarun @ritikrishu @shubhshrma @ParthS007 
Please suggest additions/updates. I guess we'll allow some freedom to the designer and see what they can come up with. This way we'll get to see what might look better on the front-end.

**Comment by shubhshrma** [comment]
@realslimshanky The flow could be like there will be an admin who will approve any project submission request from anyone else.

Another option could be that we give accounts only to moderators of all selected projects then there won't be a need for review and approval. 

Or better we could give accounts to all selected candidates in all the four tracks so that they can make a profile page of theirs to list their contributions so anyone can view. And only moderator accounts can have the capability to submit projects. 

Just my thoughts :smile:

**Comment by shubhshrma** [comment]
Some of my thoughts on the flow. It could be like there will be an admin who will approve any project submission request from anyone else. 

Another option could be that we give accounts only to moderators of all selected projects then there won't be a need for review and approval. Or better we could give accounts to all selected candidates in all the four tracks so that they can make a profile page of theirs to list their contributions so anyone can view. And only moderator accounts can have the capability to submit projects. 

Just my thoughts. Totally open to suggestions. Potential front end designers can also take ideas from this what we are aiming at.

**Comment by PrabhanshuAttri** [comment]
I think we can use [Ant Design](https://ant.design) for CRA.

**Comment by anurag-majumdar** [comment]
Ritikrishu, nicely said, will keep those points in mind. Working on a wireframe design for the first iteration and will share the images.

**Comment by highskillzz** [comment]
I think it's better we divide who makes wire frame for which. If everyone can tell what they are working on, I will choose mine

**Comment by ritikrishu** [comment]
@realslimshanky @divyamrastogi should we mark this issue assigned?

**Comment by anurag-majumdar** [comment]
I am currently working on an idea regarding the landing page, sign up and login page and list of projects screen when the user clicks on projects in the landing page.

**Comment by shubhshrma** [comment]
> controls everything, can approve submissions, all other permissions. requires authentication

@ritikrishu I am a little bit spectical about one central admin when there will already be project owners for all selected projects. Other than that, the description was perfect :smile:. Front end developers can now probably start creating things having a destination in mind.

**Comment by blenderskool** [comment]
I have been working on a few layouts. Also here are the proposed colors and fonts for the app.

Font used - [**Nunito**](https://fonts.google.com/specimen/Nunito)

https://projects.invisionapp.com/boards/m/B83LXHAGZSV

I would add all future layouts in the board above.

Do share your thoughts for the designs. Also try to maintain the above board for all designs. This will keep the them organised.

**Comment by ParthS007** [comment]
@blenderskool  They are looking cool! :smiley:

**Comment by anurag-majumdar** [comment]
I recently worked on the sign-up and login page wireframe for desktop and mobile. Colours can be changed. just to give an idea of the wireframe and how it will look.
Here are the following pics of the first iteration of the design:

![web sign up](https://user-images.githubusercontent.com/13219942/39025648-0ac22e48-4466-11e8-93de-b3e88ac5f3ba.jpg)
![mobile login and signup](https://user-images.githubusercontent.com/13219942/39025649-0afc96e6-4466-11e8-8e08-ce1bf5bbc995.jpg)
![web log in](https://user-images.githubusercontent.com/13219942/39025650-0b35f120-4466-11e8-98a1-e2908c5a77f4.jpg)

Also, here is a link to the prototype of the login page and how it can be expected to function:
https://www.youtube.com/watch?v=MKzkp1y5sXk&feature=youtu.be

The Signup and login pages for the desktop can have a background image or a predefined colour. For mobile, the login and signup screen will take the full viewport.

Let me know what you guys feel regarding this. Since, the wireframe was created with Adobe XD I decided to link the jpg files here directly.

@blenderskool and manajit.pal let me know what you guys feel too.

I am currently working on the landing page wireframe and the project list part will share that asap.

**Comment by anurag-majumdar** [comment]
Today I will be working on the prototype with wireframe regarding the flow from landing page to sign up pages. I think instead of using radio buttons we can have a separate button on the landing page requesting for sign up or login as admin or project owner. Let me know what other observations you guys feel regarding the previous design. Thanks in advance.

**Comment by blenderskool** [comment]
@anurag-majumdar  Great work! I would suggest moving the title of the app at the top. Having two login buttons on a single page is a little misleading

**Comment by blenderskool** [comment]
@ParthS007 Thanks. I have added more layouts for the user profile page and the notifications.

NOTE: I am just experimenting with many features in the UI. We may not be able to add all features, but if in the future we want to expand it, we can do so :slightly_smiling_face:

**Comment by ritikrishu** [comment]
I think we should create a zeplin (or similiar) thing and assign this issue to individuals with separate screens. They can coordinate with collaborators team and among themselves for initial branding and start picking up and working on different issues. We should not waste much time discussing this issue, instead start implementation ASAP.
@divyamrastogi @palnabarun @ParthS007 @PrabhanshuAttri @shubhshrma @realslimshanky

**Comment by blenderskool** [comment]
@ritikrishu Those are the proposed designs by me for the actual app. I have designed most of the layouts based on the data we need to display, including some of my personal ideas for the app that we can implement once the basic version is ready.

**Comment by ritikrishu** [comment]
@blenderskool it'll be nice to have the designs first and link to it. So anyone contributing to the feature will have a clear picture of what they are supposed to build. It'll also help communicating the details better while creating issues.
We'll decide on what libraries if any, we are going to use for UI components once we've a basic design set that is finalised by the whole team. This issue is not for that discussion, not yet at least!

**Comment by ParthS007** [comment]
@ritikrishu  I totally agree with you. 
@realslimshanky @shubhshrma @PrabhanshuAttri @divyamrastogi  Please give your views

**Comment by palnabarun** [comment]
@blenderskool  Great designs. There is one thing though, we can't install a project. Instead, we can view it. So, it be better if the button shows View instead of Install. I hope it makes sense.

@ritikrishu  Agreed. Let's have a call.

**Comment by blenderskool** [comment]
@palnabarun Even I thought about that. As I designed it based on @realslimshanky post-https://blog.shanky.xyz/google-udacity-scholarship.html#google-udacity-scholarship I saw Install button in the explorer hence decided to add it to the design.

We can change that to 'View' while development, I think it is pretty minor fix :slightly_smiling_face:

**Comment by PrabhanshuAttri** [comment]
@anurag-majumdar and @blenderskool Impressive work. I think we can choose @blenderskool's design. Ant design or material UI, both work for me, as long as the framework has good support and large community. This will be good for long-term support.

@blenderskool Just out of curiosity, which app did you use to create this UI? Gravit Designer or InvisionApp?

**Comment by blenderskool** [comment]
@PrabhanshuAttri Thanks. For designing the layout, I used Gravit designer. And keeping the layouts, files organized, I am using InvisionApp board.

**Comment by nikitasharma2692** [comment]
@anurag-majumdar , Great work. Just one suggestion.
On the Dashboard page(Projects list page) Showing such big filter will be a challenge for a small screens(keeping in mind lowest screen size (240 x 320)) What do you think ?

**Comment by anurag-majumdar** [comment]
@nikitasharma2692 I think that filter icon is a dummy for now and the filter feature can be added later on which will include search items and all. This part can be continued after the first workable version is shipped I guess. What do you guys feel?

Nevertheless, I don't think it will be tough to render on the smallest of devices since the planning of the filter options will be made similar to how other sites like Amazon, Flipkart, Myntra use. Our filter won't need too much options luckily.

**Comment by PrabhanshuAttri** [comment]
Continuing discussion about UI framework from #14 to avoid separate discussions. 

[Ant.Design](https://ant.design/docs/react/use-with-create-react-app) is also tailored to react. It's same as material UI, but the UI components are way closer to @blenderskool design. Also, it's aesthetically pleasing and non-monotonous like material UI.

**Comment by shubhshrma** [comment]
Thanks for the great enthusiasm on this, contributors!

There is a need of assignment of different designs to different people as this is an urgent issue. We can create a checklist in this issue or maybe separate issues for every page and get the designs done quick. Now, we need to list out all the final pages we'll have in the app. @realslimshanky @ritikrishu @ParthS007 

The primary screens that are clearly critical are as suggested by @ritikrishu :
- [ ] **Home Page**- shows all projects as thumbnails with short description.
- [ ] **Individual Project Page**- shows all details about a project- moderators, contributors etc.
- [ ] **Login/Register Page**- project owners can signup here.
- [ ] **Submit Project Page**-  detailed form for project submission by project owners.
- [ ] **Admin Panel**- admin privileges yet to be decided.

This is just the initial list and updation will be done as we progress. Front end devs please volunteer for all these so that we can get these completed and move forward to other things.

> We'll decide on what libraries if any, we are going to use for UI components once we've a basic design set that is finalised by the whole team. This issue is not for that discussion, not yet at least!

Cannot agree more :+1:

**Comment by blenderskool** [comment]
@PrabhanshuAttri I am not differentiating between the two, but what I would suggest is something like a blend between both.
One thing I like about Material design is **ripple effects**. They are much better than Ant design's hover effects. But Material design does not have a good looking input field whereas Ant design has a good range of input fields.
Since my design is based on **cards and drop shadows**, we may have to **modify** any of the design libraries we use. In fact, even if we end up using Ant design, I would strongly suggest to add some ripple effects to different parts of the app.

**Comment by blenderskool** [comment]
@shubhshrma As of now I have added the **Individual project**, basic **admin panel** (more info required to finish it) and **projects browser**(home page) layout in the design board.

@anurag-majumdar has wireframed the **authentication page** so we can have it ready soon.
I think we can start development on these screens if the designs are finalized.

**Comment by anurag-majumdar** [comment]
Thanks for the valuable feedback on the previous Wireframe iteration.
I have added a landing page to the app and some changes as suggested by @blenderskool and my fellow co-mods.

All the designs and prototypes are built using Adobe XD.

Here is the video of the prototype built so far:
https://www.youtube.com/watch?v=-agyA6J7Icc&feature=youtu.be

Landing Page: 
![web landing page](https://user-images.githubusercontent.com/13219942/39074335-8e0f69de-450e-11e8-8e25-0f04bc2e8664.jpg)

The mobile design will not be much different from this except for all items getting center aligned for the landing page and use of the side navbar for the menu items.

Web Signup Page: 
![web sign up](https://user-images.githubusercontent.com/13219942/39074380-ace6a9f8-450e-11e8-8f23-230c819f8eb1.jpg)

Web Login Page:
![web log in](https://user-images.githubusercontent.com/13219942/39074385-b58316dc-450e-11e8-8753-d47fc11512c7.jpg)

Mobile Signup/Login Page:
![mobile login and signup](https://user-images.githubusercontent.com/13219942/39074393-bf7c91e0-450e-11e8-8b83-1c760ab242db.jpg)

Regarding the signup and login pages, I decided to remove the radio buttons and instead on click of landing page, the person who wants to be an admin or project owner, the role can be set while traversing to the signup or login page. That way, it will give a better approach.

Style Guide Followed so far:
![style guide](https://user-images.githubusercontent.com/13219942/39074425-e3eee60e-450e-11e8-8460-ce1591da8f53.jpg)

As @shubhshrma pointed out, I will take on the design regarding the **projects list** page and **submit projects** page (project owner authenticated part) and share them asap.

Feel free to share your feedback on the designs.
I will continue with the pages which are left to give a more solid idea of the working of the app.

**Comment by anurag-majumdar** [comment]
Regarding blending of the look and feel for material and Ant design, I agree with @blenderskool on this. The best thing about material design is perhaps the use of shadows, which we can customize easily on this app using css.
The designs created by @blenderskool and us so far are having a sort of mix of both. Its a mix of good design practices and not necessarily Material design or Ant design only.

The ripple effect is really cool. I was checking out the effect here:
https://www.youtube.com/watch?v=FqoM_T99gvc

This video shows how to use the ripple effect. Ultimately whatever framework is decided, we have to re-create that in code by use of libraries and following the wireframe patterns.

**Comment by anurag-majumdar** [comment]
@blenderskool I liked the idea of the layouts you put out. Just in places of install I put view and view on github for individual project page. Other than that I think functionality wise things look quite good.

Thanks for the feedback regarding the usage of the same border radius.
I am sharing the desktop size mockups which are half completed as of now.
These mockups will follow most of the design strategies used by @blenderskool in terms of content.

Will complete the layout soon, as of now just sharing the layout of the content to give a picture of the application.

All Projects Page:
![web projects home page](https://user-images.githubusercontent.com/13219942/39100043-6d353650-46a1-11e8-9dbe-9381f5b025eb.png)


Individual Project Page:
![web individual project page](https://user-images.githubusercontent.com/13219942/39100047-73c22578-46a1-11e8-8764-f47d38472000.png)


Project Submission Page:
![web project submission page](https://user-images.githubusercontent.com/13219942/39100048-77d6169c-46a1-11e8-97ad-1673e1f457d3.png)


Admin Panel:
![web admin panel](https://user-images.githubusercontent.com/13219942/39100051-7ae8e13e-46a1-11e8-8618-de9a0f185df5.png)


Project Owner Panel:
![web project owner panel](https://user-images.githubusercontent.com/13219942/39100052-7f4f7e04-46a1-11e8-9268-8c7d6bd6a47f.png)

**The most important point regarding the above designs is that the font sizes and card sizes will change with viewport size. I think as per responsive design, this will make the app much easier to build from design perspective.**

I think there should not be any panel for users as they are not going to be authenticated. 
They are free to explore the app.

Also, I saw some functionalities which Admin can do as suggested by @blenderskool , i.e., changing the role from user to project owner. That can be a possibility.

One con which I can see from our designs is that we are including many extra features for the app, but that will only help us in the long run.

**I want to stress that for a minimal working app for May 14 deadline, we can go forward with the project adding, approval all which are related to projects, all information need to be added to the database of this app which can be displayed accordingly**. 

Let me know what you guys think about the layout mockups for desktop and also share your views regarding further functionalities.

**Comment by blenderskool** [comment]
@anurag-majumdar Awesome work. I have to make some minor changes to the submission form. Other than that I think we are ready for further development.

**Comment by divyamrastogi** [comment]
@blenderskool @anurag-majumdar : Awesome work guys! Really appreciate you doing this for the community.
Is it possible for you guys to add these designs on Zeplin? 
https://zeplin.io/ is a tool for UI developers that makes things easier while developing. We'll start assigning tasks after that.

**Comment by anurag-majumdar** [comment]
@divyamrastogi We need to upload our jpg files to Zeplin for reference right?

**Comment by blenderskool** [comment]
@divyamrastogi @anurag-majumdar I think we can use this https://evrybo.com it is similar to InvisionApp but completely free. What do you think?

**Comment by anurag-majumdar** [comment]
@blenderskool @divyamrastogi I am good with anything. Also @blenderskool and I were thinking if we could make a mockups folder in the existing project and add the desktop and mobile folders with the jpg files as a mockup. Apart from sharing the links this could be another additional way to give in to the project I guess.

**Comment by divyamrastogi** [comment]
@blenderskool @anurag-majumdar : Evrybo seems good for prototyping, let's go with that. But for UI development, uploading them on Zeplin will enable UI devs to find colors, dimensions and styles easy.

**Comment by anurag-majumdar** [comment]
@divyamrastogi Thanks for the invite. We can also upload it there. Currently @blenderskool and I have uploaded our images at this url: https://photos.app.goo.gl/5nm1W01LSmGwOfGh2

This is mainly Google photos where we have also posted our comments with respect to the designs.
The final design will be an amalgamation of choosing the best of both the mobile and desktop worlds.

Let us know what you guys feel. Also note that content written in paragraphs and writings can be changed as needed.

@realslimshanky @shubhshrma @ritikrishu @PrabhanshuAttri @highskillzz

**Comment by PrabhanshuAttri** [comment]
Great work guys.
Minor thing: @anurag-majumdar  I found the following mockup.
![capture](https://user-images.githubusercontent.com/4739486/39633526-e0389894-4fd5-11e8-8517-e0c0e3f6c21b.PNG)

It contains Nazi flag. This is anti-Semite and we should not associate with anything like that. Please avoid using symbols or language which is hateful to some community. Please remove it. Rest of your works fine to me. (Note: Swastika is different from Nazi flag)
@realslimshanky @shubhshrma @ritikrishu @PrabhanshuAttri @highskillzz

**Comment by realslimshanky** [comment]
Nice work guys, Closing this issue now since our project has been integrated with Material-UI in #22 and Design guidelines have been added to the documentation as well in #32.
